src/qt/pivx/topbar.cpp

```diff
@@ -541,23 +540,50 @@ void TopBar::updateBalances(const CAmount& balance, const CAmount& unconfirmedBa
     CAmount pivAvailableBalance = balance;
     // zPIV Balance
     CAmount matureZerocoinBalance = zerocoinBalance - unconfirmedZerocoinBalance - immatureZerocoinBalance;
+    bool fHaveZerocoins = zerocoinBalance > 0;
 
     // Set
     QString totalPiv = GUIUtil::formatBalance(pivAvailableBalance, nDisplayUnit);
     QString totalzPiv = GUIUtil::formatBalance(matureZerocoinBalance, nDisplayUnit, true);
+
+    // PIV
     // Top
     ui->labelAmountTopPiv->setText(totalPiv);
-    ui->labelAmountTopzPiv->setText(totalzPiv);
-
     // Expanded
     ui->labelAmountPiv->setText(totalPiv);
-    ui->labelAmountzPiv->setText(totalzPiv);
-
     ui->labelPendingPiv->setText(GUIUtil::formatBalance(unconfirmedBalance, nDisplayUnit));
-    ui->labelPendingzPiv->setText(GUIUtil::formatBalance(unconfirmedZerocoinBalance, nDisplayUnit, true));
-
     ui->labelImmaturePiv->setText(GUIUtil::formatBalance(immatureBalance, nDisplayUnit));
-    ui->labelImmaturezPiv->setText(GUIUtil::formatBalance(immatureZerocoinBalance, nDisplayUnit, true));
+
+    // Update display state and/or values for zPIV balances as necessary
+    if (fHaveZerocoins) {
+        // Set visibility of zPIV label titles/values
+        ui->typeSpacerTop->setVisible(true);
+        ui->typeSpacerExpanded->setVisible(true);
+        ui->labelAmountTopzPiv->setVisible(true);
+        ui->labelTitleAvailablezPiv->setVisible(true);
+        ui->labelAvailablezPiv->setVisible(true);
+        ui->labelTitleImmaturezPiv->setVisible(true);
+        ui->labelImmaturezPiv->setVisible(true);
+        ui->labelTitlePendingzPiv->setVisible(true);
+        ui->labelPendingzPiv->setVisible(true);
```

Comment:
No need to hide/show every single element, you can do the same only showing and hiding the container. Which will hide/show all of its child at once. Code will be shorter and cleaner.

In the dashboard zPIV balance case, it's the verticalLayout_8 the container, which should be morphed to a QWidget to be able to hide it (remember to remove the margin. Needs to be zero, morph sets the margin to 1 by default).

Other point is remember to not re update the view's visibility if it's already being shown or already hided.